tools/pubsys/src/aws/client.rs

```diff
@@ -1,159 +1,165 @@
-use async_trait::async_trait;
+use aws_config::profile::ProfileFileCredentialsProvider;
+
+use aws_config::{ConfigLoader, SdkConfig};
+use aws_sdk_ec2::{Endpoint, Region};
+use aws_sdk_sts::model::Credentials as StsCredentials;
+use aws_sdk_sts::{Client as StsClient, Credentials as SdkCredentials};
 use pubsys_config::AwsConfig;
```

Comment:
I found the client builder code very confusing and the ambiguous name of this type was a big part of why.

I would recommend either renaming this type to something like `AwsConfigOverride` or importing it differently:
```suggestion
use pubsys_config::AwsConfig as PubsysAwsConfig;
```